src/application/agora/utils.ts

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import { TwitterUserPoolingResult } from 'src/host/twitter';
+
+import { AGORA_WEBSITE_URL, TWITTER_HANDLE_TO_AGORA } from './constants';
+
+export const getProposalUrl = (proposalId: string) => {
+  return `${AGORA_WEBSITE_URL}/proposals/${proposalId}`;
+};
+
+export const getAgoraUsernameFromTwitterUsername = (handle: string) => {
+  return TWITTER_HANDLE_TO_AGORA[handle.toLowerCase()];
+};
+
+export const getDaysUntil = (dateInMs: number) => {
+  const currentDate = new Date();
+
+  // Set the time of the current date to 00:00:00 for accurate day difference calculation
+  currentDate.setHours(0, 0, 0, 0);
+
+  // Create a new date object for the target date, also set to 00:00:00
+  const targetDate = new Date(dateInMs);
+  targetDate.setHours(0, 0, 0, 0);
+
+  // Calculate the difference in milliseconds
+  const differenceInMillis = targetDate.getTime() - currentDate.getTime();
+
+  // Convert the difference from milliseconds to days
+  const differenceInDays = differenceInMillis / (1000 * 60 * 60 * 24);
+
+  return Math.round(differenceInDays);
+};
+
+export const getEndsInLabel = (daysUntil: number) => {
+  if (daysUntil < 0) {
+    return '-';
+  }
+
+  if (daysUntil === 0) {
+    return 'Ends today';
+  }
```

Comment:
since these 2 functions are same in snapshot and agora I feel temptation to move it into some reusable function inside shared module, at the same time this might change in future so I am also fine to just duplicate this logic, WDYT?

Reply:
Yeah, I was going to ask you about that. I'm not sure if the API will return only `ACTIVE` proposals, but maybe I will add handling past dates and move it to the shared module, I don't think that will cause any problems in the snapshot usage.

Reply:
What do you think about JSDoc? I like using that in the common utils. to quickly understand the function, but I know some people don't like combining that with TS. 

![image](https://github.com/idriss-crypto/browser-extensions/assets/106707979/bfdff1ab-6f9a-43c9-b02d-e094f103ce92)
![image](https://github.com/idriss-crypto/browser-extensions/assets/106707979/3ae87805-832f-4c83-802e-6f546aa6ee74)

Reply:
I like the idea about adding JSDoc! I was thinking about it but don't have experience with it. Feel free to add it wherever you see it's worth to do so.

I know that in JSDoc you can also type params/return type, should we still do it if we have ts types?

Reply:
I will also start using JSDoc

Reply:
> Yeah, I was going to ask you about that. I'm not sure if the API will return only `ACTIVE` proposals, but maybe I will add handling past dates and move it to the shared module, I don't think that will cause any problems in the snapshot usage.



> Yeah, I was going to ask you about that. I'm not sure if the API will return only `ACTIVE` proposals, but maybe I will add handling past dates and move it to the shared module, I don't think that will cause any problems in the snapshot usage.

Let's leave it as it is for now and see if api returns only active proposals and than we can adjust.